layouts/index.html

```diff
@@ -7,9 +7,9 @@
     <a name="page-top"></a>
     {{ partial "nav.html" . }}
 
-    {{ with $.Site.Data.index }}
+    {{ with .Resources.ByType "hero" }}
```

Comment:
.Resources.ByType is using MIME types and therefore we can't reuse it for fragments. In this case it needs to be plain "page" as we need the page resource.